### PR TITLE
Update on init_spark_on_yarn

### PR DESCRIPTION
**Pull request by hkvision**
- spark_conf to conf to avoid misunderstanding, as discussed in #2565 
- num_executor to num_executors to align with spark.

### CODE REVIEW

**Comment by hkvision** [comment]
jenkins passed: http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Validation/3815/

File: pyzoo/zoo/util/spark.py [hunk 1]
```diff
@@ -182,7 +182,7 @@ def _submit_opt():
                 "spark.driver.cores": driver_cores,
```
Comment (jenniew) [inline pyzoo/zoo/util/spark.py:182]: spark.driver.memory put to submit args

**Comment by jenniew** [comment]
LGTM

File: pyzoo/zoo/util/spark.py (hunk 1 above)
Comment (hkvision) [reply pyzoo/zoo/util/spark.py:183]: Updated.

**Comment by hkvision** [comment]
jenkins passed: http://10.239.47.210:18888/job/ZOO-PR-Validation/3824/
Thanks for your review and discussions @jenniew

**Pull request merged**